dm/tests/shardddl1/run.sh

```diff
@@ -711,6 +711,47 @@ function DM_MULTIPLE_ROWS() {
 	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER2_PORT
 }
 
+function DM_KEY_NOT_FOUND_CASE() {
+	check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+
+	run_sql_tidb "delete from ${shardddl}.${tb} where id=1;"
+	run_sql_tidb "delete from ${shardddl}.${tb} where id=2;"
+	run_sql_source1 "update ${shardddl1}.${tb1} set id=3 where id=1;"
+	run_sql_source2 "update ${shardddl1}.${tb1} set id=4 where id=2;"
+	run_sql_source1 "delete from ${shardddl1}.${tb1} where id=3;"
+	run_sql_source2 "delete from ${shardddl1}.${tb1} where id=4;"
+	check_log_contain_with_retry "no matching record is found to update/delete, ER_KEY_NOT_FOUND" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log
```

Comment:
and we can check worker1/worker2 log one by one, that will make test more stable